src/LiveComponent/src/ComponentWithMultiStepFormTrait.php

```diff
@@ -0,0 +1,283 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\LiveComponent;
+
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\Form\FormInterface;
+use Symfony\UX\LiveComponent\Attribute\LiveAction;
+use Symfony\UX\LiveComponent\Attribute\LiveProp;
+use Symfony\UX\LiveComponent\Storage\StorageInterface;
+use Symfony\UX\TwigComponent\Attribute\ExposeInTemplate;
+use Symfony\UX\TwigComponent\Attribute\PostMount;
+
+use function Symfony\Component\String\u;
+
+/**
+ * @author Silas Joisten <silasjoisten@proton.me>
+ * @author Patrick Reimers <preimers@pm.me>
+ * @author Jules Pietri <jules@heahprod.com>
+ */
+trait ComponentWithMultiStepFormTrait
+{
+    use ComponentWithFormTrait;
+    use DefaultActionTrait;
+
+    #[LiveProp]
+    public ?string $currentStepName = null;
+
+    /**
+     * @var string[]
+     */
+    #[LiveProp]
+    public array $stepNames = [];
+
+    public function hasValidationErrors(): bool
+    {
+        return $this->form->isSubmitted() && !$this->form->isValid();
+    }
+
+    /**
+     * @internal
+     *
+     * Must be executed after ComponentWithFormTrait::initializeForm()
+     */
+    #[PostMount(priority: -250)]
+    public function initialize(): void
+    {
+        $this->currentStepName = $this->getStorage()->get(
+            \sprintf('%s_current_step_name', self::prefix()),
+            $this->formView->vars['current_step_name'],
+        );
+
+        $this->form = $this->instantiateForm();
+
+        $formData = $this->getStorage()->get(\sprintf(
+            '%s_form_values_%s',
+            self::prefix(),
+            $this->currentStepName,
+        ));
```

Comment:
```suggestion
        $formData = $this->getStorage()->get(\sprintf('%s_form_values_%s', self::prefix(), $this->currentStepName));
```

Symfony prefers one liners to avoid merge conflicts

same for the other code parts